src/utils.c

```diff
@@ -351,7 +351,7 @@ int map_intd(char *s, char **v, int dv)
 		return atoi(s);
 	}
 	for (i = 0; v[i]; i++)
-		if (!strncasecmp(s, v[i], strlen(v[i])))
+		if (strlen(s) == strlen(v[i]) && !strncasecmp(s, v[i], strlen(v[i])))
```

Comment:
Was this tested? I am pretty sure this is not right.

Reply:
This is absolutly correct, and it works! I'm using this patch for long time.

Without it, an horrendous bug targets the current implementation: when you want to use `-l socketworks` what is enabled is `socket` because "socketworks" starts with "socket".
Please, try it!